### ISSUE TITLE
MuonBasePair object has no attribute forward_group error

### ISSUE THREAD

**Issue by robertapplin** [mantidproject/mantid#32455]
**Original reporter:** James Lord

**Describe the bug**
An error message `MuonBasePair object has no attribute forward_group error` appears when stepping to the next run after creating a phasequad.

**Reproduce this**
1. Open Frequency Domain Analysis interface
2. Load MUSR 62260
3. On Phase Table tab click `Calculate Phase Table` and then click the `+` at the bottom. Choose any name for them
4. On Transform tab, select the real and imaginary parts of the phase quad you just created.
5. Click `Calculate FFT`
6. Go to fitting tab, Add a Lorentzian and fit. (The fit will be bad)
7. Step to the next run. An error box appears `‘MuonBasePair’ object has no attribute ‘forward_group’`.

**Platform/Version (please complete the following information):**
Windows

**Comment by AnthonyLim23** [comment]
This also happens in muon analysis:

Load data
Create a phase table
Create a phasequad (the + button on the phase tab)
Change run

**Comment by AnthonyLim23** [comment]
This is coming from the corrections presenter (raise the error again instead of the warning message):
```
Traceback (most recent call last):
  File "C:\Users\BTR75544\work\dev\mantid\scripts\Muon\GUI\Common\thread_model.py", line 108, in warning
    self._exception_callback(message)
  File "C:\Users\BTR75544\work\dev\mantid\scripts\Muon\GUI\Common\corrections_tab_widget\corrections_presenter.py", line 136, in handle_thread_error
    self.view.warning_popup(error)
  File "C:\Users\BTR75544\work\dev\mantid\scripts\Muon\GUI\Common\corrections_tab_widget\corrections_view.py", line 76, in warning_popup
    warning(message, parent=self)
  File "C:\Users\BTR75544\work\dev\mantid\scripts\Muon\GUI\Common\message_box.py", line 11, in warning
    raise ValueError(error)
ValueError: 'MuonBasePair' object has no attribute 'forward_group'
```

**Comment by DavidFair** [comment]
Fixed by #32464
